### PR TITLE
Tests for ARM, adds ManagedIdentityTrackedResource

### PR DESCRIPTION
**Pull request by v-hongli1**
# Cadl Ranch Contribution Checklist:

- https://github.com/Azure/cadl-ranch/issues/605

### Supported scenarios
- ManagedIdentityTrackedResourc
  - createWithSystemAssignedOnly(put)
  - updateWithUserAssignedAndSystemAssigne(patch)
  - get

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: .changeset/fresh-cheetahs-pump.md
Comment (XiaofeiCao) [inline .changeset/fresh-cheetahs-pump.md:5]:
```suggestion
Added tests for ARM common-type's managed-identity.
```

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (XiaofeiCao) [inline packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp:28]: This is not correct. Please check the usage in the issue:
https://github.com/Azure/cadl-ranch/issues/605

```typescript
model Service is TrackedResource<ServiceProperties> {
  @doc("The name of Azure API Center service.")
  @maxLength(90)
  @minLength(1)
  @pattern("^[a-zA-Z0-9-]{3,90}$")
  @path
  @key("serviceName")
  @segment("services")
  name: string;

  ...ManagedServiceIdentityProperty;
}
```

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (XiaofeiCao) [inline packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp:44]: For SystemAssignedManagedIdentity, the response body should be something like:
```json
"identity":{
  "type": "SystemAssigned",
  "principalId": <any uuid string>,
  "tenantId": <any uuid string>
}
```

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (XiaofeiCao) [inline packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp:84]: For requests containing UserAssignedManagedIdentity, the identity property should be like:
```json
"identity":{
  "type": "SystemAssigned,UserAssigned",
  "userAssignedIdentities": {
    <user-assigned-identity-id>: {}
  }
}

```

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (XiaofeiCao) [inline packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp:92]: Response body should be like:
```json
"identity":{
  "type": "SystemAssigned,UserAssigned",
  "userAssignedIdentities": {
    "<user-assigned-identity-id>": {
      "principalId": <any-uuid-string>,
      "tenantId": <any-uuid-string>,
    },
    // this is for system-assigned
    "principalId": <any-uuid-string>,
    // this is for system-assigned
    "tenantId": <any-uuid-string>,
  }
}
```

**Comment by XiaofeiCao** [comment]
Lack one scenario: updateWithUserAssignedOnly
And make sure local live test pass.

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp]: Fxied in the new version.

**Comment by v-hongli1** [comment]
@XiaofeiCao 
> Lack one scenario: updateWithUserAssignedOnly And make sure local live test pass.
`update` cannot be repeated in tsp because the `expected path` of the `scenario` is the same.

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/main.tsp:21]: namespace should be ResourceManager instead of Arm?

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/main.tsp:31]: Is it required in this test?

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp:1]: Is there a reason we use this `identity.tsp` instead of just have it in `main.tsp`?

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp:48]: delete these `//`

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp:94]: `createWithSystemAssigned`

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp:121]: This should be the same `"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id1"`

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp:94]: Do we really need to use the `Async` op?

Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp:94]: Yes, `Arm resource operations` only include `Async` operations. 
The detail information following as: https://azure.github.io/typespec-azure/docs/next/howtos/ARM/resource-operations

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/main.tsp
Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/main.tsp:21]: The @armProviderNamespace decorator must be used to define the ARM namespace of the service.  This is best applied to the file-level namespace.

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp:94]: https://azure.github.io/typespec-azure/docs/next/howtos/ARM/resource-operations#resource-createorupdate-operations-put

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/main.tsp
Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/main.tsp]: The namespace has been changed to `Azure.ResourceManager.Models.CommonTypes.ManagedIdentity`.

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/main.tsp
Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/main.tsp]: The union `ProvisioningState` has been removed.

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp:1]: The content of `identity.tsp` have been merged into `main.tsp` and the `identity.tsp` removed.

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp]: Fixed in the new version.

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp]: Fixed in the new version.

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp]: Fixed in the new version.

File: packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp
Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/arm/models/common-types/managed-identity/identity.tsp]: `Async` op changed to `Sync`, and verification and testing successful.

File: packages/cadl-ranch-specs/cadl-ranch-summary.md
Comment (XiaofeiCao) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:785]: remove comments like this

File: packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/main.tsp
Comment (XiaofeiCao) [inline packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/main.tsp:87]: Is this needed?

File: packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/main.tsp
Comment (XiaofeiCao) [inline packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/main.tsp:138]: remove comments like this

**Comment by pshao25** [comment]
@weidongxu-microsoft none (I mean literally none) of the LRO test are supported in .net, because we have a troublesome known [issue](https://github.com/Azure/autorest.csharp/issues/4878). Therefore I cannot review.

File: packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/main.tsp
Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/main.tsp]: Fixed in the version.

**Comment by weidongxu-microsoft** [comment]
> @weidongxu-microsoft none (I mean literally none) of the LRO test are supported in .net, because we have a troublesome known [issue](https://github.com/Azure/autorest.csharp/issues/4878). Therefore I cannot review.

what are you talking about...

I asked author to use Sync.

**Comment by pshao25** [comment]
> > @weidongxu-microsoft none (I mean literally none) of the LRO test are supported in .net, because we have a troublesome known [issue](https://github.com/Azure/autorest.csharp/issues/4878). Therefore I cannot review.
> 
> what are you talking about...
> 
> I asked author to use Sync, not Async.

Didn't notice that.

@mcgallan could you have a try and prepare at .net side?
Steps:
1. Check out this PR.
2. Run eng\ApplyCadlRanch.ps1 at autorest.csharp.
3. Add these tests in autorest.csharp to see if there is any problem.
4. After this PR merged, send out PR for autorest.csharp.

File: packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts
Comment (pshao25) [inline packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts:88]: We have some problem with empty tags, which cannot be fixed in a short time. Can we add a tag here?

File: packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts
Comment (pshao25) [inline packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts:13]: Can't we just reference the SUBSCRIPTION_ID_EXPECTED above instead of putting 00000000-0000-0000-0000-000000000000 here?

File: packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts
Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts:88]: @pshao25 
Do you want to add an empty tags map in request and response data?

File: packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts
Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts]: Fixed in the new version.

File: packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts
Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts]: Add an empty tags in `main.tsp` and `mockapi.ts`.

Comment (pshao25) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts]: I mean add a tag field with items.

Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts]: image
Is this the tags you want to add?

File: .changeset/fresh-cheetahs-pump.md
Comment (weidongxu-microsoft) [inline .changeset/fresh-cheetahs-pump.md:2]: Regenerate the changeset. I don't think this be major. Should be just a patch.

File: packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/main.tsp:31]: Is there any language having problem if the TrackedResource does not have `properties` property?

File: .changeset/fresh-cheetahs-pump.md
Comment (v-hongli1) [reply .changeset/fresh-cheetahs-pump.md]: Fixed in the new version.

**Review by weidongxu-microsoft — APPROVED** [review]
You need 2 approvals for cadl-ranch PR.

**Comment by XiaofeiCao** [comment]
LGTM. Please wait for approval from another language.

**Review by XiaofeiCao — APPROVED** [review]

**Review by msyyc — APPROVED** [review]
Looks good to Python. If Java already passes these cases, I think it is OK to merge

File: packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts:104]: How's language SDK behave, when the `properties` property only contains read-only property?

Do it send the empty `properties` object, or not send `properties` at all?

Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts:104]: It send the empty properties object when the properties property only contains read-only property.

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts:104]: Yeah, I am checking for language other than Java.

If some other language does it differently, we may better to add a writable property within.

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts:104]: @v-hongli1  Or are you able to skip the check about `properties`?

Basically, do not do a full equal on body. Just check e.g. the `identities` property in the request body. This is what your test cases, not the other part of the request.

Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts:104]: @weidongxu-microsoft 
I have tried your idea, but found that the subjects must be full equality on the body to avoid causing a validation exception for the request data..

Comment (XiaofeiCao) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts:104]: @v-hongli1 I don't quite understand. Could you paste the code?

Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts:104]: @XiaofeiCao 
If it is a test case code, please refer to the following as:
```
    public void testManagedIdentityManager() {
        Map<String, String> tagsMap = new HashMap<>();
        tagsMap.put("tagKey1", "tagValue1");

        Map<String, UserAssignedIdentity> userAssignedIdentityMap = new HashMap<>();
        userAssignedIdentityMap.put(
                "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id1",
                new UserAssignedIdentity()
        );

        ManagedIdentityTrackedResource resource = manager.managedIdentityTrackedResources()
                .define("identity")
                .withRegion(Region.US_EAST)
                .withExistingResourceGroup("test-rg")
                .withProperties(new ManagedIdentityTrackedResourceProperties())
                .withIdentity(new ManagedServiceIdentity().withType(ManagedServiceIdentityType.SYSTEM_ASSIGNED))
                .withTags(tagsMap)
                .create();
        Assertions.assertEquals(ManagedServiceIdentityType.SYSTEM_ASSIGNED, resource.identity().type());

        resource = manager.managedIdentityTrackedResources().getById(resource.id());
        Assertions.assertEquals(ManagedServiceIdentityType.SYSTEM_ASSIGNED, resource.identity().type());

        resource.update().withIdentity(
                new ManagedServiceIdentity()
                        .withType(ManagedServiceIdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED)
                        .withUserAssignedIdentities(userAssignedIdentityMap))
                .apply();
        Assertions.assertEquals(ManagedServiceIdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED, resource.identity().type());
        Assertions.assertNotNull(resource.identity().userAssignedIdentities());
    }
```

Comment (XiaofeiCao) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts:104]: I mean the code in `mockapi.ts`. You mentioned that you tried not to do a full equal on body, but still can't get the verification pass?

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts:104]: You may not able to use the ready method like `req.expect.bodyEquals`

Just check the source of the func, likely do something like
```
  if (!deepEqual(request.body["identities"], expectedIdentity, { strict: true })) {
    throw new ValidationError("message", expectedIdentity, request.body["identities"]);
  }
```

Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts]: > You may not able to use the ready method like `req.expect.bodyEquals`
> 
> Just check the source of the func, likely do something like
> 
> ```
>   if (!deepEqual(request.body["identities"], expectedIdentity, { strict: true })) {
>     throw new ValidationError("message", expectedIdentity, request.body["identities"]);
>   }
> ```

Fixed following as : https://github.com/Azure/cadl-ranch/pull/609/commits/0e090c459b96f2162eac2bacb5ec89754f924b3a

File: packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts:139]: Same here? Only check "identity".

Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/mockapi.ts]: The validation of request body has been fixed to `req.expect.deepEqual(actualBody, expectedBody)`;

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM

File: packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/main.tsp:128]: Could you try in any real service, whether non-empty value is allowed?

e.g.
```
"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id1": {
  "principalId": <any uuid string>,
  "clientId": <any uuid string>
}
```
in request

Comment (v-hongli1) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/main.tsp:128]: 1. The member variables in `ManagedServiceIdentityProperty` are all read-only, so the value cannot be set when creating and updating Identity.
2. In addition, the key of `userAssignedIdentities` is the id of `managed Identity`, so I think the `principalId` and `clientId` of each usesigned managed Identity will be filled by the backend according to the id of managed Identity.
So your idea may not be realized.

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/main.tsp:128]: User can directly use the `ManagedServiceIdentityProperty` from a previous response. They don't need to set the value. The response is de-serialized by the SDK and it contains these values.

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/resource-manager/models/common-types/managed-identity/main.tsp:128]: OK, I checked the serialize code, it won't serialize this 2 properties. We are good.

```java
    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
        jsonWriter.writeStartObject();
        return jsonWriter.writeEndObject();
    }
```

**Pull request merged**
